docs/gitbook/chain-management/cascading-deployment.md

```diff
@@ -0,0 +1,85 @@
+
+Running `interbit-cli build` for a configuration file will generate an Interbit manifest file and a set of corresponding artifacts.
+
+## The Interbit Manifest File
+
+The Manifest file contains statements about how to deploy the build artifacts to a specific environment.  These files can be used to upgrade an existing environment.
+
+The files contain no unresolved aliases or inconsistencies between aliases.  Basically, there are no variables in a manifest - everything is resolved to either a chainId, or a covenantHash, or a public key.
+
+Manifest files are always named after the input config file.
+
+Manifest files are json because they contain no computation, and only act as resolution for variables.  They are statements.
+
+In addition to resolving instructions from a configuration file about your configured chains, the manifest file adds deployment instructions for a root node that manages upgrading your blockchain network should you deploy an upgraded manifest in the future.
+
+## The Root Node
+
+The root node contains a copy of the deployed manifest file and a write join to each of your high level chains. The high level chains in turn contain a write join to their children etc. This write join allows your root node to forward updates to the manifest to your chains should it receive an update. These updates will be forwarded by interbit down the line to child chains until all network updates have been applied.
+
+A configuration file must describe an acyclic network for cascading deployment to work. The network shape must be a tree.
+
+### Root Covenant
+
+The root node covenant must be a part of all covenants that wish to enforce cascading deployment. If the root node covenant is not applied to a smart contract the chain will not automatically apply updates and deployments and system upgrades must be managed in another way.
+
+## The Interbit CLI
+
+The interbit CLI can generate and deploy manifest files and watch the chains for updates to the root manifest configuration. Running `interbit-cli build` on a valid configuration will generate a deployable manifest file. Running `interbit-cli deploy` will take a manifest file and deploy a node based on the instructions in the manifest and passed through command line options.
+
+This will spin up a set of watchers that will apply any updates to the chain when a new manifest is deployed.
+
+### Root Watcher
+
+The root watcher watches the root node for CLI level update commands. If there is a difference in a newly deployed manifest compared to the previous manifest the root watcher will apply some of the changes.
+
+The root watcher
+ - deploys and applies covenants
+ - creates new chains
+ - removes chains that are no longer in the manifest
+
+### Chain Watchers
+
+One chain watcher is created for each chain that is running on a deployed node. Each chain watcher
+ - applies join changes for that chain
```

Comment:
This suggests a load on the system My help to describe what the watcher is and emphasize that it is lighweight.